File: LeetcodeC#/Program/Program/Easy/RotateArray.cs
using System.Runtime.InteropServices;

namespace DefaultNamespace;

// Given an integer array nums, rotate the array to the right by k steps, where k is non-negative.

public class RotateArray {
    public void Rotate(int[] nums, int k)
    {
        int replacee = nums[0];
        for (var i = 0; i < nums.Length; i++)
        {
            int index = (i + k) % nums.Length;
            int replaced = nums[index];
            nums[index] = replacee;
            replacee = replaced;
        }
    }
}

File: LeetcodePython/Easy/RotateArray.py
#Given an integer array nums, rotate the array to the right by k steps, where k is non-negative.

def rotate(nums, k):
    temp = nums[-k]
    for _ in range(k):
        for x in range(len(nums)):
            temp2 = nums[x]
            nums[x] = temp
            temp = temp2

print(rotate([1,2,3,4,5,6,7], 2))